src/main/java/com/commercetools/sync/commons/utils/CustomUpdateActionUtils.java

```diff
@@ -144,7 +146,18 @@
                 final String errorMessage = format(CUSTOM_FIELDS_UPDATE_ACTIONS_BUILD_FAILED,
                     resourceTypeIdGetter.apply(oldResource),
                     resourceIdGetter.apply(oldResource), exception.getMessage());
-                syncOptions.applyErrorCallback(errorMessage, exception);
+
+                if (((ParameterizedType) syncOptions.getClass().getGenericSuperclass())
+                    .getActualTypeArguments()[0].getClass().isInstance(oldResource)) {
+                    syncOptions
+                        .applyErrorCallback(new SyncException(errorMessage, exception), oldResource,
+                            newResource,
+                            null);
+                } else {
```

Comment:
What is the use case when it could happen?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/367#discussion_r249038697